module-2/backend/week_three.md

```diff
@@ -10,14 +10,42 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 1. What is the entry at the command line to create a new rails app?
+  - rails new name_of_app -T --database=postgresql --skip-turbolinks --skip-spring
 2. What do Models generally inherit from in rails?
+  - ApplicationRecord and ActiveRecord::Base
 3. What do Controllers generally inherit from in a rails project?
-4. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+  - ApplicationController and ActionController::Base
+4. How would I create a route if I wanted to see a specific horse in my routes file assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+  - Put this in routes.rb: `resources :horses, only: [:show]`
 5. What rake task is useful when looking at routes, and what information does it give you?
+  - Rake routes tells you the prefix for the path, an example path, the VERB, and the corresponding controller#action available
 6. What is an example of a route helper? When would you use them?
+  - `company_path(company)`
+  - You should use these instead of typing out the path explicitly
 7. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
+  - the `_url` returns the full url including the http
 8. What are strong params and why are the necessary?
+  - Strong params are a list of params that are allowed to be passed to the controller. This prohibits hackers from passing in additional, unwanted param values.
 9. What role does `form_for` play in helping us create our forms?
+  - It is a shorthand that actually writes HTML code for a form for us.
 10. How does `form_for` know where to submit the user's input?
-11. Create a form using a `form_for` helper to create a new `Horse`. 
+  - You have to pass it an object (e.g. @company) and it sees if the object exists already. If it does, it proceeds to the edit route. If not, it proceeds to the new route.
+11. Create a form using a `form_for` helper to create a new `Horse`.
+
+```
+<%= form_for @horse, :html => {:class => "form-horizontal center"} do |f| %>
+  <%= render partial: "shared/errors", locals: { target: @horse } %>
+
+  <div class="form-group">
```

Comment:
Like how detailed this is, but make sure you know where the form ends and the bootstrap begins. In an interview situation I would recommend putting in the bare bones form, and then asking if they use Bootstrap. If so, you can flex a little bit with this knowledge, but otherwise move on.